### ISSUE TITLE
Implement trigger.unprime() and update scheduler

### ISSUE THREAD

**Issue by chiggs** [cocotb/cocotb#16]
There are occasions where we register a callback through GPI and yet want to deregister it before the callback is made.

For example, if a coroutine yields a list of triggers it expects to be resumed if _any_ of the triggers fire.  If/when one does, the remaining triggers that didn't fire should be de-registered, otherwise we may erroneously reschedule the coroutine at some point in the future when another trigger fires.

So in summary, triggers should only be primed if coroutines are actively blocked waiting for them to fire.  If we schedule a coroutine we should un-prime any triggers that the coroutine is waiting on that didn't fire.

This should be done using the trigger.unprime() method.
